jfr/src/main/java/io/netty/handler/jfr/JfrEventHandler.java

```diff
@@ -0,0 +1,358 @@
+            event.commit();
+        }
+    }
+
+// NOTE: Occurs so often, currently remove for perfomance reasons
+//    @Override
+//    public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+//        ChannelReadCompleteEvent event = new ChannelReadCompleteEvent();
+//        event.begin();
+//        ctx.fireChannelReadComplete();
+//        if (event.shouldCommit()) {
+//            event.setChannel(ctx.channel());
+//            event.commit();
+//        }
+//    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        ChannelReadEvent event = new ChannelReadEvent();
+        event.begin();
+        ctx.fireChannelRead(msg);
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.msg = String.valueOf(msg);
+            event.commit();
+        }
+    }
+
+// NOTE: Occurs so often, currently remove for perfomance reasons
+//    @Override
+//    public void read(ChannelHandlerContext ctx) throws Exception {
+//        ReadEvent event = new ReadEvent();
+//        event.begin();
+//        ctx.read();
+//        if (event.shouldCommit()) {
+//            event.setChannel(ctx.channel());
+//            event.commit();
+//        }
+//    }
+
+    @Override
+    public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+        WriteEvent event = new WriteEvent();
+        event.begin();
+        ctx.write(msg, promise);
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.msg = String.valueOf(msg);
+            event.commit();
+        }
+    }
+
+    @Override
+    public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
+        ChannelWritabilityChangedEvent event = new ChannelWritabilityChangedEvent();
+        event.begin();
+        ctx.fireChannelWritabilityChanged();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.isWritable = ctx.channel().isWritable();
+            event.commit();
+        }
+    }
+
+    @Override
+    public void flush(ChannelHandlerContext ctx) throws Exception {
+        FlushEvent event = new FlushEvent();
+        event.begin();
+        ctx.flush();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.commit();
+        }
+    }
+
+    // Event types
+
+    @Category({ "Netty", "ChannelHandler" })
+    private abstract static class ChannelEvent extends Event {
+        @Label("Channel")
+        String channel;
+        @Label("Channel ID")
+        String channelId;
+
+        void setChannel(Channel channel) {
+            this.channel = String.valueOf(channel);
+            //TODO: More specific channel fields?
+            this.channelId = channel.id().asLongText();
+        }
+    }
+
+    @Label("Bind")
+    @Name("io.netty.channelhandler.bind")
```

Comment:
@Name should use Java naming convention for the "class", i.e io.netty.channenhandler.Bind.